internal/hcl/graph_vertex_data.go

```diff
@@ -0,0 +1,65 @@
+package hcl
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type VertexData struct {
+	logger    *logrus.Entry
+	block     *Block
+	evaluator *Evaluator
+}
+
+func (v *VertexData) ID() string {
+	return v.block.FullName()
+}
+
+func (v *VertexData) ModuleAddress() string {
+	return v.block.ModuleAddress()
+}
+
+func (v *VertexData) Evaluator() *Evaluator {
+	return v.evaluator
+}
+
+func (v *VertexData) References() []VertexReference {
+	return referencesForBlock(v.block)
+}
+
+func (v *VertexData) Evaluate() error {
```

Comment:
do we not need to lock this?

Reply:
ignore this comment - in favour of the master comment on refactoring the interface.